examples/features/functions.effekt.md

```diff
@@ -0,0 +1,51 @@
+# Functions
+
+For defining a function, you may use similar syntax for know from other languages such as Scala:
+
+```
+def fib(n: Int): Int =
+  if (n == 0 || n == 1) 1
+  else fib(n - 1) + fib(n - 2)
+```
+
+Since `fib` is a recursive function, you need to explicitly annotate it. Value parameters like `n` always need to be 
+annotated with their type.
+
+Calling functions works as you might expect:
+
+```effekt:repl
+fib(5)
+```
+
+Perhaps unusual, you can also call `fib` using Effekt's implementation of [Uniform Function Call Syntax
+](https://en.wikipedia.org/wiki/Uniform_Function_Call_Syntax#cite_note-4):
+
+```effekt:repl
+5.fib
+```
```

Comment:
I feel like this is subtly combining two different things:
1. UFCS
2. the ability to omit parens for functions without parameters 

Could we split it into two? Show `5.fib()` first and then show the parenthesis-omitting part some time later?